### DISCUSSION TITLE
Messages - fallback mechanism for messages split into multiple files

### DISCUSSION THREAD

**Discussion by vasanthid** [amannn/next-intl#1374] [Q&A]
Hi,

I'm in the process of implementing localization using `next-intl` for our project.

We would like to have our messages split into multiple files by namespaces or components as [suggested ](https://next-intl-docs.vercel.app/docs/usage/configuration#messages-split-files)in docs.

```
const messages = {
  ...(await import(`../../messages/${locale}/login.json`)).default,
  ...(await import(`../../messages/${locale}/dashboard.json`)).default
};
```

Similarly, we also want to use strings in "en/[namespace].json" as our fallback language. For any keys in a [namepace].json, if translation is not present in specified  `locale`, we like to display text from "en/[namespace].json".
Docs mention

```
import deepmerge from 'deepmerge';
 
const userMessages = (await import(`../../messages/${locale}.json`)).default;
const defaultMessages = (await import(`../../messages/en.json`)).default;
const messages = deepmerge(defaultMessages, userMessages);
```

What is the best way to combine both the scenarios to get the default from "en\login.json" for "key1" if "fr\login.json" is missing key "key1"?  Do we need to merge locale specific and default json files for each namespace in this case or are there any better approaches?

Appreciate any input and thoughts here.  Thank you.

**Comment by NanyThery** [comment]
Hi, I'm very late to the party, but it sounds that your case could be solved by deep-merging both locales before serving? I'll post a link to that.

[Messages fallback](https://next-intl.dev/docs/usage/configuration#messages-fallback)
